src/sourcegraph.d.ts

```diff
@@ -717,6 +717,14 @@ declare module 'sourcegraph' {
         ): ProviderResult<Location[]>
     }
 
+    export interface QueryTransformerProvider {
+        transformQuery(query: string): ProviderResult<string>
```

Comment:
I wrote 2 comments above about:

- not special-casing `null` return values
- using `Promise<string>` instead of `Promise<string | null | undefined>`

I see why you included null and undefined now, because `ProviderResult` adds them. But I think `ProviderResult` is only useful for things like hover providers, not for transform funcs. So you should just make this return `string | Promise<string>`.

The reason why `ProviderResult` is useful for hover/definition/reference/etc. providers is that they need to be able to return `null` to say "I have no answer for this query". But in `transformQuery`, you can say "I have no transformations to apply to this query" by just returning the original query input argument.

Another related reason you should use `string | Promise<string>` instead of `ProviderResult<string>` here is that it adds confusion if there are 3 things that all mean the same thing ("do not transform this query"):

- `query => query`
- `query => null`
- `query => undefined`

There is no need for the 2nd and 3rd ways of expressing this.